### PR TITLE
add gulp 4 and node 6 support

### PR DESCRIPTION
**Pull request by dtothefp**
#### Description
Updates `pty.js` and `node-sass` to work with Node 6. Also, added opt in gulp 4 functionality so we don't see the lame error 

```
fs: re-evaluating native module sources is not supported. If you are using the graceful-fs module, please update it to a more recent version.
```

https://laracasts.com/discuss/channels/elixir/message-re-evaluating-native-module-sources-is-not-supported

Unfortunately, we will still see this issue in the `karma` task because the `browserstacktunnel-wrapper` module had a dependency on `unzip` with old `graceful-fs` https://github.com/pghalliday/node-BrowserStackTunnel/issues/26.

Whatevvss... :neckbeard: 

#### To Test
- `nvm install 6 && npm i -g gulp-cli`
- delete the pty.js module from the nodemon package and node-sass from the webpack styles package
- `npm i` then `npm run lerna:bootstrap`
- run gulp commands as normal with gulp 4

to test gulp 3
- `nvm install 6.1 && npm i -g gulp` or if you don't want to switch node versions you can `which gulp` and delete the executable at this path, then do `npm i -g gulp`
- npm i gulp@3
- run gulp tasks with the `--gulpfile gulp3.babel.js` flag

### CODE REVIEW

**Comment by stephenkao** [comment]
I'm getting an error at the `copy` task when I try to run `gulp watch --force` (gulp v4.0.0-alpha.2, node v6.2.1):

```
[11:33:50] Starting 'copy'...
[11:33:50] 'babel' errored after 1.21 ms
[11:33:50] TypeError: Cannot read property 'name' of undefined
    at getTaskName (gulp-task-utils.js:36:19)
    at index.js:40:22
    at get-tasks.js:87:21
    at taskWrapper (/Users/stephenkao/workspace/dtothefp/build-boiler/node_modules/undertaker/lib/set-task.js:13:15)
    at bound (domain.js:280:14)
    at runBound (domain.js:293:12)
    at asyncRunner (/Users/stephenkao/workspace/dtothefp/build-boiler/node_modules/async-done/index.js:36:18)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickDomainCallback (internal/process/next_tick.js:122:9)
[11:33:50] 'watch' errored after 3.28 s
[11:33:50] The following tasks did not complete: copy
[11:33:50] Did you forget to signal async completion?
```

It looks like an empty string is getting set for the task in undertaker.

File: gulp3.babel.js [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+      );
+    }
+  });
+
+  gulp.task('default', ['build']);
+
+
+  gulp.task('test:integration', (cb) => {
+    $.sequence(
+      ['clean', 'lint'],
```
Comment (stephenkao) [inline gulp3.babel.js]: This isn't relevant to this pull request, but is there any reason we invoke the `clean` task before running integration tests?  It seems like an unnecessary step for testing.

**Comment by stephenkao** [comment]
Tested in both gulp 3 and gulp 4, and this looks good to me!  It might be worth it to add a test to verify the new `metaData` payload (especially if we end up adding more config fields to it in the future), but it's not super critical right now.

File: gulp3.babel.js (hunk 1 above)
Comment (dtothefp) [inline gulp3.babel.js]: yeh that's a mistake I've been fixing in our other repos

**Pull request merged**